reference/config_files/default_profile.rst

```diff
@@ -0,0 +1,519 @@
+so you can override some settings or environment variables for some specific package:
+
+.. code-block:: text
+   :caption: *[CONAN_HOME]/profiles/zlib_with_clang*
+
+    [settings]
+    zlib:compiler=clang
+    zlib:compiler.version=3.5
+    zlib:compiler.libcxx=libstdc++11
+    compiler=gcc
+    compiler.version=4.9
+    compiler.libcxx=libstdc++11
+
+    [buildenv]
+    zlib:CC=/usr/bin/clang
+    zlib:CXX=/usr/bin/clang++
+
+
+Your build tool will locate **clang** compiler only for the **zlib** package and **gcc** (default one) for the rest of your dependency tree.
+
+They accept patterns too, like ``-s *@myuser/*``, which means that packages that have the username "myuser" will use clang 3.5 as compiler, and gcc otherwise:
+
+.. code-block:: text
+
+    [settings]
+    *@myuser/*:compiler=clang
+    *@myuser/*:compiler.version=3.5
+    *@myuser/*:compiler.libcxx=libstdc++11
+    compiler=gcc
+    compiler.version=4.9
+    compiler.libcxx=libstdc++11
+
+Also `&` can be specified as the package name. It will apply only to the consumer conanfile (.py or .txt).
+This is a special case because the consumer conanfile might not declare a `name` so it would be impossible to reference it.
+
+.. code-block:: text
+
+    [settings]
+    &:compiler=gcc
+    &:compiler.version=4.9
+    &:compiler.libcxx=libstdc++11
+
+.. note::
+
+    If you want to override existing system environment variables, you should use the ``key=value`` syntax. If you need to pre-pend to the
+    system environment variables you should use the syntax ``key=[value]`` or ``key=[value1, value2, ...]``. A typical example is the
+    ``PATH`` environment variable, when you want to add paths to the existing system PATH, not override it, you would use:
+
+    .. code-block:: text
+
+        [buildenv]
+        PATH=[/some/path/to/my/tool]
+
+.. _profiles_buildenv:
+
+[buildenv]
+++++++++++
+
+Available since: `1.35.0 <https://github.com/conan-io/conan/releases/tag/1.35.0>`_
+
+.. important::
+
+    The use of this ``[buildenv]`` section requires using the ``VirtualBuildEnv`` generator in your recipe,
+    or putting the configuration ``tools.env.virtualenv:auto_use=True`` in your profile.
+
+
+This profile section is aimed to be the replacement of the legacy ``[buildenv]`` one. It's more powerful, and it is able to
+apply some additional operators to each variable declared when you're composing profiles or even local variables:
+
+* ``+=`` == ``append``: appends values at the end of the existing value.
+* ``=+`` == ``prepend``: puts values at the beginning of the existing value.
+* ``=!`` == ``unset``: gets rid of any variable value.
+
+Another essential point to mention is the possibility of defining variables as `PATH` ones by simply putting ``(path)`` as
+the prefix of the variable. It is useful to automatically get the append/prepend of the `PATH` in different systems
+(Windows uses ``;`` as separation, and UNIX ``:``).
```

Comment:
how does it work in case of cross-compiling? it's not crystal clear if it just uses `os.name` or `self.settings.os`.